var/spack/repos/builtin/packages/py-torch-geometric/package.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class PyTorchGeometric(PythonPackage):
+    """PyTorch Geometric (PyG) is a geometric deep learning extension
+    library for PyTorch.  It consists of various methods for deep
+    learning on graphs and other irregular structures, also known as
+    geometric deep learning, from a variety of published papers. In
+    addition, it consists of an easy-to-use mini-batch loader for many
+    small and single giant graphs, multi gpu-support, a large number
+    of common benchmark datasets (based on simple interfaces to create
+    your own), and helpful transforms, both for learning on arbitrary
+    graphs as well as on 3D meshes or point clouds."""
+
+    homepage = "https://github.com/rusty1s/pytorch_geometric"
+    url      = "https://github.com/rusty1s/pytorch_geometric/archive/1.6.0.tar.gz"
+
+    version('1.6.0', sha256='7d5231cdcc2ebd4444f406cbf1537eb49bf90ab6f446eaf1b7af5cdbe105f3c9')
+
+    depends_on('py-torch', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-tqdm', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-networkx', type=('build', 'run'))
+    depends_on('py-scikit-learn', type=('build', 'run'))
+    depends_on('py-numba', type=('build', 'run'))
+    depends_on('py-requests', type=('build', 'run'))
+    depends_on('py-pandas', type=('build', 'run'))
+    depends_on('py-rdflib', type=('build', 'run'))
+    depends_on('py-googledrivedownloader', type=('build', 'run'))
+    depends_on('py-h5py~mpi', type=('build', 'run'))
```

Comment:
I don't see this dependency listed

Reply:
I've taken to shutting off mpi support in hdf5-related stuff when it comes in contact with torch, as when users engage the parallel dataloader (which uses fork()) mpi tends to want to crash in nonobvious ways - I think I'm also seeing this in tensorflow now as well...

Reply:
But does this package even depend on `py-h5py`?

Reply:
yes...?
```
coltrane (Downloads/pytorch_geometric-1.6.0):grep -r "h5py" *
docker/Dockerfile:RUN conda install -y h5py=2.8.0 \
docker/Dockerfile:RUN pip install h5py-cache==1.0
setup.py:    'h5py',
torch_geometric/datasets/dynamic_faust.py:import h5py
torch_geometric/datasets/dynamic_faust.py:        fm = h5py.File(self.raw_paths[0], 'r')
torch_geometric/datasets/dynamic_faust.py:        ff = h5py.File(self.raw_paths[1], 'r')
torch_geometric/datasets/s3dis.py:import h5py
torch_geometric/datasets/s3dis.py:            f = h5py.File(osp.join(self.raw_dir, filename))

```

Reply:
Ah, sorry, I'm blind. It's even in the `setup.py`, I'm not sure why I didn't notice.